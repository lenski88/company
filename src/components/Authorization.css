.Authorization {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  border: 0.2em solid  #fbfc55;
  padding: 3em;
  margin-top:25vh
}

.Authorization .inputText {
    border-color: #fbfc55;
    background-color: #fbfc55 ;
    color:#0301a8;
    font-family:  'Press Start 2P', cursive;
    padding: 0.3em;
}
