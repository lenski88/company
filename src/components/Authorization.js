import React from "react";
import PropTypes from "prop-types";

import "./Authorization.css";

class Authorization extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.array.isRequired /* массив сотрудников */,
    login:
      PropTypes.bool.isRequired /* прошла ли авторизация; по умолчанию false */,
    cbGetUser: PropTypes.func,
  };

  state = {
    login: this.props.login,
    nameUser: "",
    errName: "",
    button: false,
  };

  nameUser = (eo) => {
    this.setState({
      nameUser: eo.target.value,
    });

    let arrayNames = this.props.employes;
    let name = eo.target.value;
    let isLogin = arrayNames.some((i) => {
      return name === i.login;
    });

    if (!isLogin) {
      this.setState({
        errName: "Нет такого сотрудника",
      });
    } else {
      this.setState({
        errName: "",
      });
    }
  };

  getUser = () => {
    if (!this.state.errName && !this.state.errPassword) {
      let arrayNames = this.props.employes;
      let name = this.state.nameUser;
      let user = arrayNames.find((i) => {
        return name === i.login;
      });
      this.props.cbGetUser(user);
      this.setState({
        nameUser: "",
        login: this.props.login,
      });
    }
  };

  render() {
    return (
      !this.state.login && (
        <div className={!this.props.login?"Authorization": "NoAuthorization"}>
          <label>Введите логин:</label>
          <br />
          <input
            className="inputText"
            type="text"
            name="nameUser"
            value={this.state.nameUser}
            onChange={this.nameUser}
            autoFocus
          ></input>
          <br />
          <span>{this.state.errName}</span>
          <br />
          <br />
          {this.state.nameUser && (
            <input
              className="button"
              type="button"
              onPointerDown={this.getUser}
              value="[Войти]"
              disabled={this.state.button}
            ></input>
          )}
        </div>
      )
    );
  }
}

export default Authorization;
