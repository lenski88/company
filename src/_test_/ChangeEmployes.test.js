"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { employes } from "./Authorization.test";

import ChangeEmployes from "../components/ChangeEmployee";

/* {
    id: 40,
    login: "kulakovI40",
    status: 0,
    level: 2,
    name: "Кулаков Иван Иванович",
    email: "kulakov@example.com",
    phone: "294389052",
    department: "Складское хозяйство",
    position: "Начальник отдела",
    task: [],
  } */

test("проверка правильности значений инпутов при открытии карточки", () => {
  let component = renderer.create(
    <ChangeEmployes employes={employes} idEmp={employes[0].id} indexEmp={0} />
  );
  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let level = component.root.find((el) => el.props.name == "level");
  expect(level.props.value).toBe(2);

  let email = component.root.find((el) => el.props.name == "email");
  expect(email.props.value).toBe("kulakov@example.com");

  let phone = component.root.find((el) => el.props.name == "phone");
  expect(phone.props.value).toBe("294389052");

  let department = component.root.find((el) => el.props.name == "department");
  expect(department.props.value).toBe("Складское хозяйство");

  let position = component.root.find((el) => el.props.name == "position");
  expect(position.props.value).toBe("Начальник отдела");
});

test("проверка ввода", () => {
  let component = renderer.create(
    <ChangeEmployes employes={employes} idEmp={employes[0].id} indexEmp={0} />
  );
  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let level = component.root.find((el) => el.props.name == "level");
  level.props.onChange({ target: { value: 2 } });

  let email = component.root.find((el) => el.props.name == "email");
  email.props.onChange({ target: { value: "dad@errr.com" } });

  let phone = component.root.find((el) => el.props.name == "phone");
  phone.props.onChange({ target: { value: '"295559052"' } });

  let department = component.root.find((el) => el.props.name == "department");
  department.props.onChange({ target: { value: "Руководство" } });

  let position = component.root.find((el) => el.props.name == "position");
  position.props.onChange({ target: { value: "Рукамиводитель" } });

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test("проверка некорректного ввода", () => {
  let component = renderer.create(
    <ChangeEmployes employes={employes} idEmp={employes[0].id} indexEmp={0} />
  );
  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let level = component.root.find((el) => el.props.name == "level");
  level.props.onChange({ target: { value: 2 } });

  let email = component.root.find((el) => el.props.name == "email");
  email.props.onChange({ target: { value: "daderrr.com" } });

  let phone = component.root.find((el) => el.props.name == "phone");
  phone.props.onChange({ target: { value: '"29555ппп9052"' } });

  let department = component.root.find((el) => el.props.name == "department");
  department.props.onChange({ target: { value: "Руководство" } });

  let position = component.root.find((el) => el.props.name == "position");
  position.props.onChange({ target: { value: "рукамиводитель" } });

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
