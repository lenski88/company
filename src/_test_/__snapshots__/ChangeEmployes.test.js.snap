// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка Authorization 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка Authorization 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="sdfasdf"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="saveliev47"
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка ввода 1`] = `
<div
  className="ChangeEmp"
>
  <h2
    style={
      Object {
        "color": "#fbfc55 ",
        "marginLeft": "1vw",
        "textAlign": "center",
      }
    }
  >
    Внесение изменений в карточку сотрудника
     
    Кулаков Иван Иванович
  </h2>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      autoFocus={true}
      className="inputText"
      name="level"
      onChange={[Function]}
      value={2}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onChange={[Function]}
      type="text"
      value="kulakov@example.com"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onChange={[Function]}
      type="text"
      value="294389052"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onChange={[Function]}
      value="Складское хозяйство"
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onChange={[Function]}
      type="text"
      value="Начальник отдела"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Сохранить]"
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Выйти без сохранения]"
  />
</div>
`;

exports[`проверка ввода 2`] = `
<div
  className="ChangeEmp"
>
  <h2
    style={
      Object {
        "color": "#fbfc55 ",
        "marginLeft": "1vw",
        "textAlign": "center",
      }
    }
  >
    Внесение изменений в карточку сотрудника
     
    Кулаков Иван Иванович
  </h2>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      autoFocus={true}
      className="inputText"
      name="level"
      onChange={[Function]}
      value={2}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onChange={[Function]}
      type="text"
      value="dad@errr.com"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onChange={[Function]}
      type="text"
      value="\\"295559052\\""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      Содержит 9 цифр
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onChange={[Function]}
      value="Руководство"
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onChange={[Function]}
      type="text"
      value="Рукамиводитель"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Выйти без сохранения]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка нажатия на кнопку войти 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка некорректного ввода 1`] = `
<div
  className="ChangeEmp"
>
  <h2
    style={
      Object {
        "color": "#fbfc55 ",
        "marginLeft": "1vw",
        "textAlign": "center",
      }
    }
  >
    Внесение изменений в карточку сотрудника
     
    Кулаков Иван Иванович
  </h2>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      autoFocus={true}
      className="inputText"
      name="level"
      onChange={[Function]}
      value={2}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onChange={[Function]}
      type="text"
      value="kulakov@example.com"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onChange={[Function]}
      type="text"
      value="294389052"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onChange={[Function]}
      value="Складское хозяйство"
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onChange={[Function]}
      type="text"
      value="Начальник отдела"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Сохранить]"
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Выйти без сохранения]"
  />
</div>
`;

exports[`проверка некорректного ввода 2`] = `
<div
  className="ChangeEmp"
>
  <h2
    style={
      Object {
        "color": "#fbfc55 ",
        "marginLeft": "1vw",
        "textAlign": "center",
      }
    }
  >
    Внесение изменений в карточку сотрудника
     
    Кулаков Иван Иванович
  </h2>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      autoFocus={true}
      className="inputText"
      name="level"
      onChange={[Function]}
      value={2}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onChange={[Function]}
      type="text"
      value="daderrr.com"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      Формат ввода: login@example.com
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onChange={[Function]}
      type="text"
      value="\\"29555ппп9052\\""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      Содержит 9 цифр
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onChange={[Function]}
      value="Руководство"
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onChange={[Function]}
      type="text"
      value="рукамиводитель"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      Первая буква заглавная; только кириллица
    </span>
  </label>
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Выйти без сохранения]"
  />
</div>
`;

exports[`проверка правильности значений инпутов при открытии карточки 1`] = `
<div
  className="ChangeEmp"
>
  <h2
    style={
      Object {
        "color": "#fbfc55 ",
        "marginLeft": "1vw",
        "textAlign": "center",
      }
    }
  >
    Внесение изменений в карточку сотрудника
     
    Кулаков Иван Иванович
  </h2>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      autoFocus={true}
      className="inputText"
      name="level"
      onChange={[Function]}
      value={2}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onChange={[Function]}
      type="text"
      value="kulakov@example.com"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onChange={[Function]}
      type="text"
      value="294389052"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onChange={[Function]}
      value="Складское хозяйство"
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onChange={[Function]}
      type="text"
      value="Начальник отдела"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Сохранить]"
  />
  <input
    className="button"
    onPointerDown={[Function]}
    type="button"
    value="[Выйти без сохранения]"
  />
</div>
`;

exports[`проверка рендера кнопки при верном логине 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка рендера кнопки при верном логине 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;
